app/modules/site_settings/resources.py

```diff
@@ -173,155 +172,35 @@ class MainConfigurationDefinition(Resource):
     Site Setting Definitions
     """
 
-    def _insert_houston_definitions(self, data):
-        for key in SiteSetting.get_setting_keys():
-            data['response']['configuration'][
-                key
-            ] = SiteSetting.get_as_edm_definition_format(key)
-        return data
-
     def get(self, path):
-        if not is_extension_enabled('edm'):
-            data = {'response': {'configuration': {}}}
-            return self._insert_houston_definitions(data)
-
-        data = SiteSetting.get_edm_configuration_as_edm_format(
-            'configurationDefinition.data', path
-        )
-
-        from app.modules.ia_config_reader import IaConfig
-
-        species_json = None
-        if SiteSetting.is_block_key(path):
-            species_json = data['response']['configuration']['site.species']
-        elif path == 'site.species':
-            species_json = data['response']
-        if species_json is not None:
-            ia_config_reader = IaConfig()
-            species = ia_config_reader.get_configured_species()
-            if not isinstance(species_json['suggestedValues'], list):
-                species_json['suggestedValues'] = ()
-            for (
-                sn
-            ) in species:  # only adds a species that is not already in suggestedValues
-                needed = True
-                for sv in species_json['suggestedValues']:
-                    if sv.get('scientificName', None) == sn:
-                        needed = True
-                if needed:
-                    details = ia_config_reader.get_frontend_species_summary(sn)
-                    if details is None:
-                        details = {}
-                    species_json['suggestedValues'].insert(
-                        0,
-                        {
-                            'scientificName': sn,
-                            'commonNames': [details.get('common_name', sn)],
-                            'itisTsn': details.get('itis_id'),
-                        },
-                    )
-
-        if SiteSetting.is_block_key(path):
-            data = self._insert_houston_definitions(data)
-
-        # TODO also traverse private here FIXME
-        return data
+        # Currently only support getting all definitions as one
+        if SiteSetting.is_rest_block_key(path):
+            return SiteSetting.get_all_rest_definitions()
+        else:
+            abort(400, 'Can only get the full block of definitions, not individual items')
```

Comment:
I need to talk to Ben about this. If you look at how this is used in the frontend, I think there needs to be a ew API that does what the FE wants in one message rather than 2

Reply:
I'm suggesting:

```python
@api.route('/definition/main/block')
class MainConfigurationDefinition(Resource):
    def get(self):
        return SiteSetting.get_all_rest_definitions()
```

The API would return 400 anyway for anything else, right?

Reply:
Oh I guess we need to add `@api.route('/definition/main/__bundle_setup')`

Reply:
I do see what you are saying but yeah, it is possible to read the block via `/definition/main/` `/definition/main/block` and `/definition/main/_bundle_setup`
But as I say, the goal is to combine the block configuration and the definition into one message at which point the existing definition and block configuration endpoints will be rertired so I'm not keen on optimising them.